# Classificação de Gênero Simples
Vamos classificar o gênero

#Sobre o conjunto de dados
As 10 colunas neste modelo são:

    - ID: identificador único para cada indivíduo no conjunto
    - de dados Sexo: masculino ou feminino
    - Idade: idade em anos
    - Altura (cm): altura em centímetros
    - Peso (kg): peso em quilogramas
    - Ocupação: profissão ou cargo
    - Nível de educação: nível mais alto de educação atingido
    - Estado civil: solteiro, casado, divorciado, viúvo, etc.
    - Renda (USD): renda anual em dólares americanos
    - Cor favorita: cor favorita do indivíduo (esta coluna é apenas por diversão e não tem significado real)

Esse conjunto de dados pode ser usado para várias tarefas de classificação e análise de dados, como prever o gênero ou o nível de renda de um indivíduo com base em sua ocupação, nível de educação e outros fatores demográficos.

In [1]:
import pandas as pd
import matplotlib.pyplot as plt
import numpy as np
import seaborn as sns
import warnings
warnings.filterwarnings("ignore")

genero = pd.read_csv(r'DtScience\gender.csv')
display(genero)

Gender   Age   Height (cm)   Weight (kg)             Occupation  \
0       male    32           175            70      Software Engineer   
1       male    25           182            85   Sales Representative   
2     female    41           160            62                 Doctor   
3       male    38           178            79                 Lawyer   
4     female    29           165            58       Graphic Designer   
..       ...   ...           ...           ...                    ...   
126   female    32           170            64                  Nurse   
127     male    38           176            79        Project Manager   
128   female    27           162            55       Graphic Designer   
129     male    33           175            77   Sales Representative   
130   female    29           164            57     Software Developer   

         Education Level  Marital Status   Income (USD)  Favorite Color  \
0        Master's Degree         Married          75000            Blue   
1      Bachelor's Degree          Single          45000           Green   
2       Doctorate Degree         Married         120000          Purple   
3      Bachelor's Degree          Single          90000             Red   
4     Associate's Degree          Single          35000          Yellow   
..                   ...             ...            ...             ...   
126   Associate's Degree          Single          60000          Orange   
127    Bachelor's Degree         Married          90000           Black   
128   Associate's Degree          Single          55000           Green   
129    Bachelor's Degree         Married          80000          Yellow   
130    Bachelor's Degree          Single          65000            Blue   

     Unnamed: 9  
0           NaN  
1           NaN  
2           NaN  
3           NaN  
4           NaN  
..          ...  
126         NaN  
127         NaN  
128         NaN  
129         NaN  
130         NaN  

[131 rows x 10 columns]

In [2]:
genero = genero.drop("Unnamed: 9", axis=1)
display(genero)

Gender   Age   Height (cm)   Weight (kg)             Occupation  \
0       male    32           175            70      Software Engineer   
1       male    25           182            85   Sales Representative   
2     female    41           160            62                 Doctor   
3       male    38           178            79                 Lawyer   
4     female    29           165            58       Graphic Designer   
..       ...   ...           ...           ...                    ...   
126   female    32           170            64                  Nurse   
127     male    38           176            79        Project Manager   
128   female    27           162            55       Graphic Designer   
129     male    33           175            77   Sales Representative   
130   female    29           164            57     Software Developer   

         Education Level  Marital Status   Income (USD)  Favorite Color  
0        Master's Degree         Married          75000            Blue  
1      Bachelor's Degree          Single          45000           Green  
2       Doctorate Degree         Married         120000          Purple  
3      Bachelor's Degree          Single          90000             Red  
4     Associate's Degree          Single          35000          Yellow  
..                   ...             ...            ...             ...  
126   Associate's Degree          Single          60000          Orange  
127    Bachelor's Degree         Married          90000           Black  
128   Associate's Degree          Single          55000           Green  
129    Bachelor's Degree         Married          80000          Yellow  
130    Bachelor's Degree          Single          65000            Blue  

[131 rows x 9 columns]

In [3]:
genero.columns = ["Sexo", "Idade", "Altura (cm)", "Peso (kg)", "Profissão", "Escolaridade", "Estado_civil", "Renda (USD)", "Cor_favorita"]

In [4]:
genero.Sexo.replace(to_replace= [' male', ' female'], value= ['male', 'female'], inplace= True)

In [5]:
genero.Profissão.replace(to_replace=[' Teacher', ' Nurse', ' Writer', ' Project Manager', ' Marketing Specialist', 
' Lawyer', ' Doctor', ' Graphic Designer', ' Engineer', ' Architect', ' Sales Representative', ' IT Manager'
, ' Business Analyst', ' CEO', ' Accountant', ' Analyst', ' Marketing Specialist', ' Software Engineer',
 ' Business Consultant'], value= ['Teacher', 'Nurse', 'Writer',
'Project Manager', 'Marketing Specialist', 'Lawyer', 'Doctor', 'Graphic Designer', 'Engineer', 'Architect', 
'Sales Representative', 'IT Manager', 'Business Analyst', 'CEO', 'Accountant', 'Analyst', 'Marketing Specialist',
'Software Engineer', 'Business Consultant'], inplace= True)

In [6]:
genero.Cor_favorita.replace(to_replace= [' Blue', ' Purple', ' Green', ' Red', ' Yellow', ' Orange', ' Grey', ' Pink', 
' Black'], value=['Blue', 'Purple', 'Green', 'Red', 'Yellow', 'Orange', 'Grey', 'Pink', 'Black'], inplace= True)

In [7]:
genero.Escolaridade.replace(to_replace= [" Bachelor's Degree", " Associate's Degree", " Master's Degree", " Doctorate Degree"],
value= ["Bachelor's Degree", "Associate's Degree", "Master's Degree", "Doctorate Degree"], inplace= True)

In [8]:
genero.Estado_civil.replace(to_replace= [' Married', ' Single', ' Divorced', ' Widowed'], 
value= ['Married', 'Single', 'Divorced', 'Widowed'], inplace= True)

In [9]:
genero.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 131 entries, 0 to 130
Data columns (total 9 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   Sexo          131 non-null    object
 1   Idade         131 non-null    int64 
 2   Altura (cm)   131 non-null    int64 
 3   Peso (kg)     131 non-null    int64 
 4   Profissão     131 non-null    object
 5   Escolaridade  131 non-null    object
 6   Estado_civil  131 non-null    object
 7   Renda (USD)   131 non-null    int64 
 8   Cor_favorita  131 non-null    object
dtypes: int64(4), object(5)
memory usage: 9.3+ KB


In [10]:
print(genero["Sexo"].value_counts())
print(genero["Sexo"].value_counts(normalize=True).map("{:.1%}".format))

Sexo
male      68
female    63
Name: count, dtype: int64
Sexo
male      51.9%
female    48.1%
Name: proportion, dtype: object


In [11]:
import plotly.express as px
for coluna in genero.columns:
    grafico = px.histogram(genero, x=coluna, color="Sexo", text_auto=True)
    grafico.show()

# Conclusão:

- Temos 131 pessoas sendo 68 do sexo masculino e 63 sexo feminino, sendo: 
    - Masculino:    51.9%
    - Feminino:     48.1%
#
### Idade:

- O maior grupo de mulheres tem entre 26 a 33 anos de idade.

- O maior grupo de homens tem entre 34 e 45 anos de idade, tendo apenas 3 homens que superam essa idade.
#
### Altura:

- Nesse grupo das MULHERES a MENOR altura é de 160 cm e a MAIOR altura é de 174 cm.

- No grupo dos HOMENS a MENOR altura é 175 cm e a MAIOR altura é 194 cm.
#
### Peso:

- Nesse grupo das MULHERES o MENOR peso é de 50 kg e o MAIOR peso é de 69 kg.

- No grupo dos HOMENS o MENOR peso é de 70 kg e o MAIOR peso é de 94 kg.
#
### Profissão:

- Temos 8 profissões que tem a mesma quantidade de profissionais, que são 9:

    - Das 8 profissões, apenas 2 tem os homens e mulheres trabalhando na mesma área:

        - Doutor, sendo 6 mulheres e 3 homens.
        - Gestor de projeto, sendo 6 homens e 3 muheres.

    - Das 8 profissões, 4 delas só trabalham mulheres e 2 apenas homens:
        
        - Mulheres: Escritora, Enfermeira, Designer gráfico e especialista em marketing.

        - Homens: Advogado e Engenheiro.

    - A profissão que tem mais profissionais tem 12 pessoas sendo 10 mulheres e 2 homens, atuando como professores.
#
### Escolaridade:

- Temos 78 pessoas (59.5%) que fizeram Bacharelado, 42 homens e 36 mulheres.

- Só as mulheres fizeram Graduação sendo 18 delas (13.7%).

- 20 fizeram o Mestrado (15.3%) tendo 17 homens e 3 mulheres.

- E 15 fizeram o Doutorado (11.5%) com 9 homens e 6 mulheres.
#
### Estado Civil:

- Temos maior volume em Casados (65 ou 49.6%) e Solteiros (59 ou 45.0%) e um menor volume de Divorciados (5 ou 3.8%) e Viúvos (2 ou 1.5%).

    - Casados: 16 mulheres e 49 homens
    - Solteiros: 47 mulheres e 12 homens
    - Divorciados: 0 mulheres e 5 homens
    - Viúvos: 0 mulheres e 2 homens
#
### Renda (USD):

- A Renda mais comum é de 50k - 99k , com 79 pessoas recebendo essa renda 44 mulheres e 35 homens

- As Rendas mais incomuns são de 150k - 299k que é recebida apenas por 3 homens e de 500k - 549k que também é recebida apenas por 3 homens.

- A menor renda possível é entre 0 - 49k com 15 pessoas recebendo essa renda 13 mulheres e 2 homens.
#
### Cor Favorita:

- A cor mais apreciada tanto pelas mulheres quanto os homens é o Verde com 21 votos, 11 dos homens e 10 das mulheres, essa cor tem uma opinião mais balanceada entre os dois generos. 
    
    - Assim como o Vermelho e o Roxo com 17 votos cada um, com 9 votos das mulheres e 8 dos homens para a cor Vermelha e 9 votos dos homens e 8 das mulheres para a cor Roxa.

    - Tivemos também a cor Cinza que teve 8 votos, foi a cor com menor número de votos, mas teve uma igualdade no número de votos com com 4 votos para cada lado.

- A cor Azul também tem o mesmo número de votos que a cor Verde, mas ela teve mais votos dos homens, sendo 13 votos cotra os 8 votos das mulheres.

    - Como é o caso das cores Preto e Laranja que tem o mesmo número de votos 12 mas tem uma preferência maior dos homens, sendo a cor Preta com 10 votos dos homens e 2 das mulheres e a cor Laranja com 7 votos dos homens e 5 das mulheres.

- Temos as cores Amarelo e Rosa que teve mais votos das mulheres com uma grande divergência, a cor Amarela teve mais votos que a cor Rosa, com 16 votos contra 7.

    - Dos 16 votos na cor Amarela tivemos 12 votos das mulheres e 4 dos homens.

    - Dos 7 votos na cor Rosa tivemos 5 votos das mulheres e 2 dos homens.

